### PR TITLE
[Data Objects] Remove o_ prefixed of system properties and database columns 

### LINKED ISSUE #7341: [Data Objects] Remove `o_` prefixed of system properties and database columns
Add (maybe an optional) compatibility layer to the data objects views (e.g. `object_MyClass`. 

Also consider this BC layer: #12769

### PR DESCRIPTION
**Pull request by mcop1**
Resolves #7341

### CODE REVIEW

**Comment by fashxp** [comment]
@mcop1  do we need to provide a BC layer in Pimcore 10.5 for that ... especially in API calls where javascript might be depending on `o_id` vs. `id` in the payload.
For example like the get object data call [here](https://github.com/pimcore/pimcore/pull/13308/files#diff-223386524f4ff18dc1e37479f7e783a5e463cce8417be7519db1a05185549f1aL825)?

File: models/DataObject/ClassDefinition/Dao.php [hunk 1]
```diff
@@ -114,15 +114,15 @@ public function update()
         $objectView = 'object_' . $this->model->getId();
 
         // create object table if not exists
-        $protectedColumns = ['oo_id', 'oo_classId', 'oo_className'];
+        $protectedColumns = ['oo_id', 'oclassId', 'oo_className'];
```
Comment (mcop1) [inline models/DataObject/ClassDefinition/Dao.php]:
```suggestion
        $protectedColumns = ['oo_id', 'oo_classId', 'oo_className'];
```

File: bundles/CoreBundle/src/Migrations/Version20221003115124.php [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+final class Version20221003115124 extends AbstractMigration
+{
+    public function getDescription(): string
+    {
+        return 'Rename o_id columns to id.';
+    }
+
+    public function up(Schema $schema): void
+    {
+        $this->addSql("ALTER TABLE pimcore.objects CHANGE o_id id int(11) unsigned auto_increment NOT NULL;");
```
Comment (aashan10) [inline bundles/CoreBundle/src/Migrations/Version20221003115124.php]: Should we be adding the table name like `pimcore.object`? Isn't the database name configurable?

Comment (brusch) [reply bundles/CoreBundle/src/Migrations/Version20221003115124.php]: DB-name can be just removed as not needed.

Comment (mcop1) [reply bundles/CoreBundle/src/Migrations/Version20221003115124.php]: I´ll remove the database name

**Comment by mcop1** [comment]
> @mcop1 do we need to provide a BC layer in Pimcore 10.5 for that ... especially in API calls where javascript might be depending on `o_id` vs. `id` in the payload. For example like the get object data call [here](https://github.com/pimcore/pimcore/pull/13308/files#diff-223386524f4ff18dc1e37479f7e783a5e463cce8417be7519db1a05185549f1aL825)?

Just for completeness: 
PHP: There is a bc method which returns the version dependent db column name: https://github.com/pimcore/pimcore/issues/13512
JS: We can use something like `foo['o_id'] ?? foo['id']` as we did with childs/children.

File: bundles/AdminBundle/src/Controller/Admin/DataObject/DataObjectController.php [hunk 3]
```diff
@@ -1366,22 +1366,22 @@ protected function updateIndexesOfObjectSiblings(DataObject\AbstractObject $upda
...
-                            With cte As (SELECT o_index, o_id FROM ' . $list->getDao()->getTableName() . ' WHERE o_parentId = ? AND o_id != ? AND o_type IN (\''.implode(
+                            With cte As (SELECT index, id FROM ' . $list->getDao()->getTableName() . ' WHERE parentId = ? AND id != ? AND type IN (\''.implode(
                     "','", [
                         DataObject::OBJECT_TYPE_OBJECT,
                         DataObject::OBJECT_TYPE_VARIANT,
                         DataObject::OBJECT_TYPE_FOLDER,
                     ]
-                ).'\') ORDER BY o_index LIMIT '. $updatedObject->getParent()->getChildAmount() .')
-                            SELECT @n := IF(@n = ? - 1,@n + 2,@n + 1) AS newIndex, o_id
+                ).'\') ORDER BY indexLIMIT '. $updatedObject->getParent()->getChildAmount() .')
```
Comment (fashxp) [inline bundles/AdminBundle/src/Controller/Admin/DataObject/DataObjectController.php]: there, a space after `index` is necessary.

File: bundles/AdminBundle/src/Controller/Admin/DataObject/DataObjectController.php [hunk 4]
```diff
@@ -1366,22 +1366,22 @@ protected function updateIndexesOfObjectSiblings(DataObject\AbstractObject $upda
             $db->executeStatement(
                 'UPDATE '.$list->getDao()->getTableName().' o,
                     (
-                        SELECT newIndex, o_id
+                        SELECT newIndex, id
                         FROM (
-                            With cte As (SELECT o_index, o_id FROM ' . $list->getDao()->getTableName() . ' WHERE o_parentId = ? AND o_id != ? AND o_type IN (\''.implode(
+                            With cte As (SELECT index, id FROM ' . $list->getDao()->getTableName() . ' WHERE parentId = ? AND id != ? AND type IN (\''.implode(
```
Comment (fashxp) [inline bundles/AdminBundle/src/Controller/Admin/DataObject/DataObjectController.php]: `index` needs to be quoted ... it is a reserved word in mysql

File: bundles/AdminBundle/src/Controller/Admin/DataObject/DataObjectController.php [hunk 5]
```diff
@@ -1059,9 +1059,9 @@ public function deleteAction(Request $request): JsonResponse
             $parentObject = DataObject::getById((int) $request->get('id'));
 
             $list = new DataObject\Listing();
-            $list->setCondition('o_path LIKE ' . $list->quote($list->escapeLike($parentObject->getRealFullPath()) . '/%'));
+            $list->setCondition('path LIKE ' . $list->quote($list->escapeLike($parentObject->getRealFullPath()) . '/%'));
```
Comment (fashxp) [inline bundles/AdminBundle/src/Controller/Admin/DataObject/DataObjectController.php]: `path` is a reserved word and should be quoted

Comment (fashxp) [reply bundles/AdminBundle/src/Controller/Admin/DataObject/DataObjectController.php]: also consider for other appearances

File: bundles/AdminBundle/src/Controller/Admin/DataObject/DataObjectController.php (hunk 4 above)
Comment (fashxp) [reply bundles/AdminBundle/src/Controller/Admin/DataObject/DataObjectController.php]: also consider for other appearances

File: bundles/AdminBundle/src/Controller/Admin/DataObject/DataObjectController.php [hunk 6]
```diff
@@ -1390,8 +1390,8 @@ protected function updateIndexesOfObjectSiblings(DataObject\AbstractObject $upda
             );
 
             $siblings = $db->fetchAllAssociative(
-                'SELECT o_id, o_modificationDate, o_versionCount, o_key, o_index FROM objects'
-                ." WHERE o_parentId = ? AND o_id != ? AND o_type IN ('object', 'variant','folder') ORDER BY o_index ASC",
+                'SELECT id, modificationDate, versionCount, key, index FROM objects'
+                ." WHERE parentId = ? AND id != ? AND type IN ('object', 'variant','folder') ORDER BY indexASC",
```
Comment (fashxp) [inline bundles/AdminBundle/src/Controller/Admin/DataObject/DataObjectController.php]: space after `index` missing

File: bundles/CoreBundle/src/Migrations/Version20221003115124.php [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+                $tableName = $table->getName();
+                if (str_starts_with($tableName, "object_brick_") ||
+                    str_starts_with($tableName, "object_metadata_") ||
+                    str_starts_with($tableName, "object_collection_")) {
+                    $this->addSql("ALTER TABLE $tableName CHANGE o_id id int(10) unsigned DEFAULT 0 NOT NULL;");
+                } else if (str_starts_with($tableName, "object_classificationstore_")) {
+                    $this->addSql("ALTER TABLE $tableName CHANGE o_id id int(11) unsigned NOT NULL;");
+                }
+            }
+        }
```
Comment (fashxp) [inline bundles/CoreBundle/src/Migrations/Version20221003115124.php:60]: all product index tables are missing ... which have additional columns like `o_virtualProductId`, `o_virtualProductActive`, `o_classId`, `o_parentId`, `o_type`

Comment (fashxp) [reply bundles/CoreBundle/src/Migrations/Version20221003115124.php:60]: we also need to find a way how to "update" or "migrate" eventually existing elasticsearch indices ... or add upgrade notes, that they need to be rebuild...

File: models/DataObject/AbstractObject.php [hunk 8]
```diff
@@ -235,36 +233,21 @@ abstract class AbstractObject extends Model\Element\AbstractElement
...
-    }
+    protected ?array $properties = null;
 
     /**
      * {@inheritdoc}
      */
     protected function getBlockedVars(): array
     {
-        $blockedVars = ['o_hasChildren', 'o_versions', 'o_class', 'scheduledTasks', 'o_parent', 'parent', 'omitMandatoryCheck'];
+        $blockedVars = ['hasChildren', 'versions', 'class', 'scheduledTasks', 'parent', 'parent', 'omitMandatoryCheck'];
```
Comment (fashxp) [inline models/DataObject/AbstractObject.php:243]: up to now, `hasChildren`, `siblings`, `hasSiblings`, `childrenSortBy`, `childrenSortOrder`, `versionCount`, `dirtyLanguages` and `dirtyFields` were not reserved ... that means, there might be classes with such fields. 
we need at least add that to the upgrade notes and require the devs to rename/migrate these fields before updating to Pimcore 11

File: bundles/CoreBundle/src/Migrations/Version20221003115124.php (hunk 7 above)
Comment (fashxp) [reply bundles/CoreBundle/src/Migrations/Version20221003115124.php:60]: we also need to update or regenerate all `object_*` views

File: models/DataObject/ClassDefinition/Data/AdvancedManyToManyRelation.php [hunk 9]
```diff
@@ -264,12 +264,12 @@ public function getDataForEditmode(mixed $data, DataObject\Concrete $object = nu
...
+                $identifier = 'id';
                 if ($targetType == 'object') {
-                    $pathCol = 'o_path';
-                    $keyCol = 'o_key';
-                    $typeCol = 'o_type';
-                    $className = ', o_ClassName className';
+                    $pathCol = 'path';
+                    $keyCol = 'key';
+                    $typeCol = 'type';
+                    $className = ', className className';
```
Comment (fashxp) [inline models/DataObject/ClassDefinition/Data/AdvancedManyToManyRelation.php]: what is that?

File: models/Element/Service.php [hunk 10]
```diff
@@ -625,13 +611,13 @@ public static function minimizePropertiesForEditmode(array $props): array
             //$p = object2array($p);
             $allowedProperties = [
                 'key',
-                'o_key',
+                'key',
```
Comment (fashxp) [inline models/Element/Service.php]: double entries...

File: models/Element/Tag/Dao.php [hunk 11]
```diff
@@ -231,7 +231,7 @@ public function getElementsForTag(
         $map = [
             'document' => ['documents', 'id', 'type', '\Pimcore\Model\Document'],
             'asset' => ['assets', 'id', 'type', '\Pimcore\Model\Asset'],
-            'object' => ['objects', 'o_id', 'o_type', '\Pimcore\Model\DataObject\AbstractObject'],
+            'object' => ['objects', 'id', 'type', '\Pimcore\Model\DataObject\AbstractObject'],
```
Comment (fashxp) [inline models/Element/Tag/Dao.php]: we don't need that map anymore then, right?

Comment (fashxp) [reply models/Element/Tag/Dao.php]: at least `id` and `type`

**Review by fashxp — CHANGES_REQUESTED** [review]

**Comment by fashxp** [comment]
Things to test
- [x] Test grid config with o_ prefixes
- [x] Check version compare
- [x] Check vouchers & onlineshop orders
- [ ] Check elasticsearch index
- [x] Check internal search
- [ ] Check saving of localized fields
- [x] Check tags
- [x] Check delete user

File: models/DataObject/AbstractObject.php (hunk 8 above)
Comment (mcop1) [reply models/DataObject/AbstractObject.php:243]: Added an entry to the upgrade notes

File: models/Element/Tag/Dao.php (hunk 11 above)
Comment (mcop1) [reply models/Element/Tag/Dao.php]: Removed `id` and `type` from the array and adapted the code.

File: models/DataObject/ClassDefinition/Data/AdvancedManyToManyRelation.php (hunk 9 above)
Comment (mcop1) [reply models/DataObject/ClassDefinition/Data/AdvancedManyToManyRelation.php]: `className` was used as an alias in the following sql. This shouldn´t be necessary anymore. I also adapted the code a bit to eliminate the redundant variable assignments.

File: bundles/AdminBundle/src/Controller/Admin/DataObject/DataObjectController.php (hunk 5 above)
Comment (mcop1) [reply bundles/AdminBundle/src/Controller/Admin/DataObject/DataObjectController.php]: `path` isn´t a reserved word for column names, right? If we quote `path` anyway, then we should quote `type` too.

Comment (mcop1) [reply bundles/AdminBundle/src/Controller/Admin/DataObject/DataObjectController.php]: Solution: quote both

File: bundles/CoreBundle/src/Migrations/Version20221003115124.php (hunk 7 above)
Comment (mcop1) [reply bundles/CoreBundle/src/Migrations/Version20221003115124.php:60]: A hint for product index tables and elastic search indexes will be added to the upgrade notes.
`object_*` views will be regenerated by classes-rebuild in a separate migration -> https://github.com/pimcore/pimcore/issues/12859

File: models/DataObject/AbstractObject.php (hunk 8 above)
Comment (fashxp) [reply models/DataObject/AbstractObject.php:243]: but we also need to add them here in the blockedVars list, right?

Comment (fashxp) [reply models/DataObject/AbstractObject.php:243]: actually not here, but [here](https://github.com/pimcore/pimcore/blob/815ec901d89af3f57f458012993357f269c39eef/models/DataObject/ClassDefinition/Data.php#L77) and [here](https://github.com/pimcore/pimcore/blob/f25e567a626c7c6e52f53846d9ed9a5a9b374721/bundles/AdminBundle/public/js/pimcore/object/classes/data/data.js#L18)

Comment (mcop1) [reply models/DataObject/AbstractObject.php:243]: Added them to the forbidden names array

File: bundles/AdminBundle/src/Controller/Admin/DataObject/DataObjectController.php [hunk 12]
```diff
@@ -1059,9 +1059,9 @@ public function deleteAction(Request $request): JsonResponse
             $parentObject = DataObject::getById((int) $request->get('id'));
 
             $list = new DataObject\Listing();
-            $list->setCondition('o_path LIKE ' . $list->quote($list->escapeLike($parentObject->getRealFullPath()) . '/%'));
+            $list->setCondition('`path` LIKE ' . $list->quote($list->escapeLike($parentObject->getRealFullPath()) . '/%'));
             $list->setLimit((int)$request->get('amount'));
-            $list->setOrderKey('LENGTH(o_path)', false);
+            $list->setOrderKey('LENGTH(path)', false);
```
Comment (fashxp) [inline bundles/AdminBundle/src/Controller/Admin/DataObject/DataObjectController.php]: quote

File: bundles/AdminBundle/src/Controller/Admin/DataObject/DataObjectController.php [hunk 13]
```diff
@@ -1881,8 +1881,8 @@ public function copyInfoAction(Request $request): JsonResponse
             if ($object->hasChildren(DataObject::$types)) {
                 // get amount of children
                 $list = new DataObject\Listing();
-                $list->setCondition('o_path LIKE ' . $list->quote($list->escapeLike($object->getRealFullPath()) . '/%'));
-                $list->setOrderKey('LENGTH(o_path)', false);
+                $list->setCondition('`path` LIKE ' . $list->quote($list->escapeLike($object->getRealFullPath()) . '/%'));
+                $list->setOrderKey('LENGTH(path)', false);
```
Comment (fashxp) [inline bundles/AdminBundle/src/Controller/Admin/DataObject/DataObjectController.php]: quote

File: bundles/AdminBundle/src/Controller/Admin/TagsController.php [hunk 14]
```diff
@@ -316,14 +316,14 @@ public function getBatchAssignmentJobsAction(Request $request, EventDispatcherIn
...
     {
         $childrenList = new \Pimcore\Model\DataObject\Listing();
-        $condition = 'o_path LIKE ?';
+        $condition = '`path` LIKE ?';
         if (!$this->getAdminUser()->isAdmin()) {
             $userIds = $this->getAdminUser()->getRoles();
             $userIds[] = $this->getAdminUser()->getId();
             $condition .= ' AND (
-                (SELECT `view` FROM users_workspaces_object WHERE userId IN (' . implode(',', $userIds) . ') and LOCATE(CONCAT(o_path,o_key),cpath)=1  ORDER BY LENGTH(cpath) DESC LIMIT 1)=1
+                (SELECT `view` FROM users_workspaces_object WHERE userId IN (' . implode(',', $userIds) . ') and LOCATE(CONCAT(path,key),cpath)=1  ORDER BY LENGTH(cpath) DESC LIMIT 1)=1
```
Comment (fashxp) [inline bundles/AdminBundle/src/Controller/Admin/TagsController.php]: quote

File: bundles/AdminBundle/src/Controller/Admin/TagsController.php [hunk 15]
```diff
@@ -316,14 +316,14 @@ public function getBatchAssignmentJobsAction(Request $request, EventDispatcherIn
...
                     OR
-                (SELECT `view` FROM users_workspaces_object WHERE userId IN (' . implode(',', $userIds) . ') and LOCATE(cpath,CONCAT(o_path,o_key))=1  ORDER BY LENGTH(cpath) DESC LIMIT 1)=1
+                (SELECT `view` FROM users_workspaces_object WHERE userId IN (' . implode(',', $userIds) . ') and LOCATE(cpath,CONCAT(path,key))=1  ORDER BY LENGTH(cpath) DESC LIMIT 1)=1
```
Comment (fashxp) [inline bundles/AdminBundle/src/Controller/Admin/TagsController.php]: quote

File: bundles/AdminBundle/src/Helper/GridHelperService.php [hunk 16]
```diff
@@ -310,16 +310,16 @@ public function getFilterCondition(string $filterJson, ClassDefinition $class, $
                         } else {
                             $conditionPartsFilters[] = $field->getFilterCondition($filter['value'] ?? null, $operator, ['brickPrefix' => ($tablePrefix ? $tablePrefix . '.' : null)]);
                         }
-                    } elseif (in_array('o_' . $filterField, $systemFields)) {
+                    } elseif (in_array($filterField, $systemFields)) {
                         // system field
                         if ($filterField == 'fullpath') {
-                            $conditionPartsFilters[] = 'concat(o_path, o_key) ' . $operator . ' ' . $db->quote('%' . $filter['value'] . '%');
+                            $conditionPartsFilters[] = 'concat(path, key) ' . $operator . ' ' . $db->quote('%' . $filter['value'] . '%');
```
Comment (fashxp) [inline bundles/AdminBundle/src/Helper/GridHelperService.php]: quote

File: bundles/AdminBundle/src/Helper/GridHelperService.php [hunk 17]
```diff
@@ -310,16 +310,16 @@ public function getFilterCondition(string $filterJson, ClassDefinition $class, $
...
                         } elseif ($filterField == 'key') {
-                            $conditionPartsFilters[] = 'o_key ' . $operator . ' ' . $db->quote('%' . $filter['value'] . '%');
+                            $conditionPartsFilters[] = 'key ' . $operator . ' ' . $db->quote('%' . $filter['value'] . '%');
```
Comment (fashxp) [inline bundles/AdminBundle/src/Helper/GridHelperService.php]: quote

File: bundles/AdminBundle/src/Helper/GridHelperService.php [hunk 18]
```diff
@@ -528,7 +528,7 @@ public function prepareListingForGrid(array $requestParams, string $requestedLan
                 if (array_key_exists($orderKey, $colMappings)) {
                     $orderKey = $colMappings[$orderKey];
                 } elseif ($orderKey === 'fullpath') {
-                    $orderKey = 'CAST(CONCAT(o_path, o_key) AS CHAR CHARACTER SET utf8) COLLATE utf8_general_ci';
+                    $orderKey = 'CAST(CONCAT(path, key) AS CHAR CHARACTER SET utf8) COLLATE utf8_general_ci';
```
Comment (fashxp) [inline bundles/AdminBundle/src/Helper/GridHelperService.php]: quote

File: bundles/AdminBundle/src/Helper/GridHelperService.php [hunk 19]
```diff
@@ -563,20 +563,20 @@ public function prepareListingForGrid(array $requestParams, string $requestedLan
...
-            $conditionFilters[] = '(o_path = ' . $quotedPath . ' OR o_path LIKE ' . $quotedWildcardPath . ')';
+            $conditionFilters[] = '(`path` = ' . $quotedPath . ' OR `path` like ' . $quotedWildcardPath . ')';
         }
 
         if (!$adminUser->isAdmin()) {
             $userIds = $adminUser->getRoles();
             $userIds[] = $adminUser->getId();
             $conditionFilters[] = ' (
-                                                    (select list from users_workspaces_object where userId in (' . implode(',', $userIds) . ') and LOCATE(CONCAT(o_path,o_key),cpath)=1  ORDER BY LENGTH(cpath) DESC LIMIT 1)=1
+                                                    (select list from users_workspaces_object where userId in (' . implode(',', $userIds) . ') and LOCATE(CONCAT(path,key),cpath)=1  ORDER BY LENGTH(cpath) DESC LIMIT 1)=1
```
Comment (fashxp) [inline bundles/AdminBundle/src/Helper/GridHelperService.php]: quote

File: bundles/AdminBundle/src/Helper/GridHelperService.php [hunk 20]
```diff
@@ -563,20 +563,20 @@ public function prepareListingForGrid(array $requestParams, string $requestedLan
...
                                                     OR
-                                                    (select list from users_workspaces_object where userId in (' . implode(',', $userIds) . ') and LOCATE(cpath,CONCAT(o_path,o_key))=1  ORDER BY LENGTH(cpath) DESC LIMIT 1)=1
+                                                    (select list from users_workspaces_object where userId in (' . implode(',', $userIds) . ') and LOCATE(cpath,CONCAT(path,key))=1  ORDER BY LENGTH(cpath) DESC LIMIT 1)=1
```
Comment (fashxp) [inline bundles/AdminBundle/src/Helper/GridHelperService.php]: quote

File: models/DataObject/AbstractObject/Dao.php [hunk 21]
```diff
@@ -174,7 +174,7 @@ public function updateChildPaths(string $oldPath): ?array
 
             //update object child paths
             // we don't update the modification date here, as this can have side-effects when there's an unpublished version for an element
-            $this->db->executeQuery('update objects set o_path = replace(o_path,' . $this->db->quote($oldPath . '/') . ',' . $this->db->quote($this->model->getRealFullPath() . '/') . "), o_userModification = '" . $userId . "' where o_path like " . $this->db->quote(Helper::escapeLike($oldPath) . '/%') . ';');
+            $this->db->executeQuery('update objects set `path` = replace(path,' . $this->db->quote($oldPath . '/') . ',' . $this->db->quote($this->model->getRealFullPath() . '/') . "), userModification = '" . $userId . "' where `path` like " . $this->db->quote(Helper::escapeLike($oldPath) . '/%') . ';');
```
Comment (fashxp) [inline models/DataObject/AbstractObject/Dao.php]: quote

File: models/DataObject/AbstractObject/Dao.php [hunk 22]
```diff
@@ -206,7 +206,7 @@ public function getCurrentFullPath(): ?string
         $path = null;
 
         try {
-            $path = $this->db->fetchOne('SELECT CONCAT(o_path,`o_key`) as o_path FROM objects WHERE o_id = ?', [$this->model->getId()]);
+            $path = $this->db->fetchOne('SELECT CONCAT(path,`key`) as `path` FROM objects WHERE id = ?', [$this->model->getId()]);
```
Comment (fashxp) [inline models/DataObject/AbstractObject/Dao.php]: quote

File: models/DataObject/AbstractObject/Dao.php [hunk 23]
```diff
@@ -419,7 +419,7 @@ public function getChildAmount(?array $objectTypes = [DataObject::OBJECT_TYPE_OB
      */
     public function getTypeById(int $id): array
     {
-        $t = $this->db->fetchAssociative('SELECT o_type,o_className,o_classId FROM objects WHERE o_id = ?', [$id]);
+        $t = $this->db->fetchAssociative('SELECT type,className,classId FROM objects WHERE id = ?', [$id]);
```
Comment (fashxp) [inline models/DataObject/AbstractObject/Dao.php]: quote

File: models/DataObject/Data/ElementMetadata/Dao.php [hunk 24]
```diff
@@ -50,13 +50,13 @@ public function save(DataObject\Concrete $object, string $ownertype, string $own
     public function load(DataObject\Concrete $source, int $destinationId, string $fieldname, string $ownertype, string $ownername, string $position, int $index, string $destinationType = 'object'): ?DataObject\Data\ElementMetadata
     {
         if ($destinationType == 'object') {
-            $typeQuery = " AND (type = 'object' or type = '')";
+            $typeQuery = " AND (type = 'object' or `type` = '')";
```
Comment (fashxp) [inline models/DataObject/Data/ElementMetadata/Dao.php]: quote

File: models/DataObject/Data/ObjectMetadata/Dao.php [hunk 25]
```diff
@@ -58,9 +58,9 @@ protected function getTablename(DataObject\Concrete $object): string
 
     public function load(DataObject\Concrete $source, int $destinationId, string $fieldname, string $ownertype, string $ownername, string $position, int $index, string $destinationType = 'object'): ?DataObject\Data\ObjectMetadata
     {
-        $typeQuery = " AND (type = 'object' or type = '')";
+        $typeQuery = " AND (type = 'object' or `type` = '')";
```
Comment (fashxp) [inline models/DataObject/Data/ObjectMetadata/Dao.php]: quote

**Review by fashxp — CHANGES_REQUESTED** [review]

**Comment by fashxp** [comment]
More findings: 

- [x] empty parent id in frontend 
image

- [x] event objects title is always set to `Event`
image

**Comment by mcop1** [comment]
> More findings:
> 
>     * [x]  empty parent id in frontend
>       image
> 
>     * [x]  event objects title is always set to `Event`
>       image

Problems fixed. Second one was a bit tricky, since the title from the data object class definition was set into the title of the actual object.

**Review by fashxp — APPROVED** [review]

**Pull request merged**